public-ui/src/app/data-browser/views/genomic-view/components/variant-search.component.tsx

```diff
@@ -43,45 +41,58 @@ export class VariantSearchComponent extends React.Component<Props, State> {
         };
     }
 
-    search = _.debounce(() => this.getSearchSize(), 1000);
+    search = _.debounce(() => this.getVariantSearch(), 1000);
 
     // life cycle hook
     componentDidMount() {
         this.getSearchSize();
     }
 
     handleChange(val) {
-        this.setState({ searchWord: val});
+        this.setState({ searchWord: val });
         this.search(val);
     }
 
     getSearchSize() {
         genomicsApi().getVariantSearchResultSize(this.state.searchWord).then(
             result => {
-                this.setState({searchResultSize: result, searchSizeLoading: false});
+                this.setState({
+                    searchResultSize: this.state.searchWord !== '' ? result : 0,
+                    searchSizeLoading: false
+                });
             }
         ).catch(e => {
             console.log(e, 'error');
             this.setState({ searchSizeLoading: false });
         });
     }
 
+    getVariantSearch() {
+        this.getSearchSize();
+        genomicsApi().searchVariants(this.state.searchWord).then(
+            result => {
+                this.props.onSearchReturn(result);
+            }
+        );
```

Comment:
I think it makes sense to also move this call and the `searchWord` state var up to `GenomicSearchComponent` since we'll need the search term to call `getVariantSearchResultSize`. Then we can pass `searchWord` to this component as a prop and change the `onSearchReturn` prop to pass changes to `searchTerm` up to the parent.

Hopefully that makes sense, let me know if you want to do a call to discuss/clarify any of the changes.